spring-kafka/src/main/java/org/springframework/kafka/listener/DeadLetterPublishingRecoverer.java

```diff
@@ -825,25 +863,17 @@ private String getStackTraceAsString(Throwable cause) {
 	 * in the sent record.
 	 * @return the header names.
 	 * @since 2.7
+	 * @deprecated since 3.0.9 - provide a supplier in the constructor instead
```

Comment:
But I don't see that ctor arg - only a setter.
Should this be fixed respectively then?

Reply:
Sorry, that was from an earlier iteration, but I couldn't make backwards compatibility work that way. Will fix.